src/mapping/Mapping.cpp

```diff
@@ -87,6 +87,31 @@ bool Mapping::requireGradient() const
   return _requireGradient;
 }
 
+void Mapping::map(int inputDataID,
+                  int outputDataID)
+{
+  PRECICE_ASSERT(_hasComputedMapping);
+  PRECICE_ASSERT(input()->getDimensions() == output()->getDimensions(),
+                 input()->getDimensions(), output()->getDimensions());
+  PRECICE_ASSERT(getDimensions() == output()->getDimensions(),
+                 getDimensions(), output()->getDimensions());
+  PRECICE_ASSERT(input()->data(inputDataID)->getDimensions() == output()->data(outputDataID)->getDimensions(),
+                 input()->data(inputDataID)->getDimensions(), output()->data(outputDataID)->getDimensions());
+  PRECICE_ASSERT(input()->data(inputDataID)->values().size() / input()->data(inputDataID)->getDimensions() == static_cast<int>(input()->vertices().size()),
+                 input()->data(inputDataID)->values().size(), input()->data(inputDataID)->getDimensions(), input()->vertices().size());
+  PRECICE_ASSERT(output()->data(outputDataID)->values().size() / output()->data(outputDataID)->getDimensions() == static_cast<int>(output()->vertices().size()),
+                 output()->data(outputDataID)->values().size(), output()->data(outputDataID)->getDimensions(), output()->vertices().size());
+
+  if (hasConstraint(CONSERVATIVE)) {
+    mapConservative(inputDataID, outputDataID);
+  } else {
+    mapConsistent(inputDataID, outputDataID);
+  }
+  if (hasConstraint(Mapping::SCALEDCONSISTENT)) {
+    scaleConsistentMapping(inputDataID, outputDataID);
```

Comment:
I'm not a big fan of this structure, with an if coming after an else. Wouldn't it be cleaner to have something like 

```cpp
if (hasConstraint(CONSERVATIVE)) {
mapConservative(...)
} else if (hasConstraint(CONSISTENT) {
mapConsisten(...);
} else if {hasConstraint(SCALEDCONSISTENT) {
mapConsistent(...);
scaleConsistentMapping(...);
}
```

Also, we have `CONSERVATIVE` but `Mapping::SCALEDCONSISTENT`. Prefix isn't consistent.

Reply:
Yes, I adapted it to the structure you proposed here.

Reply:
... and fixed the inconsistency.